src/server/api/routers/assignment/read.tsx

```diff
@@ -0,0 +1,22 @@
+import { z } from "zod";
+import { protectedProcedure } from "@/server/api/trpc";
+import { TRPCError } from "@trpc/server";
+
+export const readAssignments = protectedProcedure
+  .input(z.object({ classId: z.string() }))
+  .query(async ({ ctx, input }) => {
+    try {
+      return await ctx.db.assignment.findMany({
+        where: {
+          deletedAt: null,
+          ...(input.classId && { classId: input.classId }),
+        },
+      });
+    } catch (error) {
+      throw new TRPCError({
+        code: "INTERNAL_SERVER_ERROR",
+        message: "Failed to fetch assignments.",
+        cause: error,
+      });
+    }
+  });
```

Comment:
add more read queries, 
1. one to look at all the deleted assignments for a given class
2.  one which gives output for a particular assignment by id, 
3. add one with releaseData
4. add one with due date
5. add one which is generic to the data provided
these will be findMany